### DISCUSSION TITLE
26.5 Drop

### DISCUSSION THREAD

**Discussion by njr0** [google/comprehensive-rust#1265] [General]
I thought you missed a trick here. I think the point is made more strongly if you either don't `drop(a)` at the end or create another variable that you don't drop. I changed it to:

```rust
fn main() {
    let a = Droppable { name: "a" };
    let aa = Droppable {name: "aa"};
    {
        let b = Droppable { name: "b" };
        {
            let c = Droppable { name: "c" };
            let d = Droppable { name: "d" };
            println!("Exiting block B");
        }
        println!("Exiting block A");
    }
    drop(a);
    println!("Exiting main");
}
```
so it prints
```
Exiting block B  
Dropping d  
Dropping c  
Exiting block A  
Dropping b  
Dropping a  
Exiting main  
Dropping aa
```
which seems to make the point more strongly, to me.

**Comment by djmitche** [comment]
Rather than adding a new variable, maybe we should remove the `drop` entirely from the code in the slide, and recommend adding it inside the nested `let`, to show that you can drop a variable in a nested scope. WDYT?

**Comment by njr0** [comment]
You mean remove all the `drop`s? Or just the last one.

I think removing the last one would be fine. I don't see any advantage in removing the earlier ones, but I guess that would be OK too...

**Comment by djmitche** [comment]
The current example looks like
```
fn main() {
    let a = Droppable { name: "a" };
    {
        let b = Droppable { name: "b" };
        {
            let c = Droppable { name: "c" };
            let d = Droppable { name: "d" };
            println!("Exiting block B");
        }
        println!("Exiting block A");
    }
    drop(a);
    println!("Exiting main");
}
```
so there's only one `drop`. Or are we talking about different things?

**Comment by njr0** [comment]
You're right, of course. Apologies, I wasn't looking at the code when I replied. So yes, I agree.
